cmip6_downscaling/methods/common/containers.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class BBox:
+    """Lat/lon bounding box"""
+
+    latmin: float = -90.0
+    latmax: float = 90.0
+    lonmin: float = -180.0
+    lonmax: float = 180
+
+    @property
+    def lat_slice(self) -> slice:
+        return slice(float(self.latmax), float(self.latmin))
+
+    @property
+    def lon_slice(self) -> slice:
+        return slice(float(self.lonmin), float(self.lonmax))
+
+    def __str__(self) -> str:
+        return f'BBox([{self.latmin}, {self.latmax}, {self.lonmin}, {self.lonmax}])'
+
+
+@dataclass
+class TimePeriod:
+    """Time period object"""
+
+    start: str = None
+    stop: str = None
+
+    @property
+    def time_slice(self):
+        return slice(self.start, self.stop)
+
+    def __str__(self) -> str:
+        return f'TimePeriod([{self.start}, {self.stop}])'
+
+
+@dataclass
+class CMIP6Experiment:
+
+    model: str
+    scenario: str
+    member: str
+
+
+@dataclass
+class RunParameters:
+    """Run parameters object"""
+
+    method: str
+    obs: str
+    model: str
+    scenario: str
+    variable: str
+    latmin: str
+    latmax: str
+    lonmin: str
+    lonmax: str
```

Comment:
should these be floats?

Reply:
https://github.com/carbonplan/cmip6-downscaling/pull/140